File: Source/Camping Mod 2.0/common/rikmuld/camping/misc/potions/PotionBleeding.java
package rikmuld.camping.misc.potions;

import net.minecraft.client.Minecraft;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.potion.Potion;
import net.minecraft.util.ResourceLocation;
import rikmuld.camping.core.lib.PotionInfo;
import rikmuld.camping.core.lib.TextureInfo;
import rikmuld.camping.core.register.ModDamageSources;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

public class PotionBleeding extends Potion {

	public PotionBleeding(String name)
	{
		super(PotionInfo.id(name), false, 9643043);
		setPotionName(PotionInfo.name(name));
		setIconIndex(0, 0);
	}

	@Override
	@SideOnly(Side.CLIENT)
	public boolean hasStatusIcon()
	{
		Minecraft.getMinecraft().renderEngine.bindTexture(new ResourceLocation(TextureInfo.SPRITE_POTION));
		return true;
	}

	@Override
	public boolean isReady(int par1, int par2)
	{
		int k = 60 >> par2;
		return k > 0? (par1 % k) == 0:true;
	}

	@Override
	public void performEffect(EntityLivingBase entity, int amplifier)
	{
		entity.attackEntityFrom(ModDamageSources.bleeding, (entity.getMaxHealth() / 20) * (amplifier + 1));
	}
}

File: scala/com/rikmuld/camping/features/blocks/trap/PotionBleeding.scala
package com.rikmuld.camping.features.blocks.trap

import com.rikmuld.camping.Library.{DamageInfo, TextureInfo}
import net.minecraft.client.Minecraft
import net.minecraft.entity.EntityLivingBase
import net.minecraft.potion.Potion
import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.relauncher.{Side, SideOnly}

object PotionBleeding {
  final val TEXTURE =
    new ResourceLocation(TextureInfo.SPRITE_POTION)

  val bleedingSource =
    new DamageSourceBleeding(DamageInfo.BLEEDING)
}

class PotionBleeding(name: String) extends Potion(false, 9643043) {
  setPotionName(name)
  setRegistryName(name)
  setIconIndex(0, 0)

  @SideOnly(Side.CLIENT)
  override def hasStatusIcon: Boolean = {
    Minecraft.getMinecraft.renderEngine.bindTexture(PotionBleeding.TEXTURE)
    true
  }

  override def isReady(par1: Int, par2: Int): Boolean = {
    val k = 60 >> par2
    if (k > 0) (par1 % k) == 0 else true
  }

  override def performEffect(entity: EntityLivingBase, amplifier: Int): Unit =
    entity.attackEntityFrom(PotionBleeding.bleedingSource, (entity.getMaxHealth / 20) * (amplifier + 1))
}